text_extensions_for_pandas/array/tensor.py

```diff
@@ -224,6 +263,14 @@ def __repr__(self):
     def __str__(self):
         return self._tensor.__str__()
 
+    def _values_for_factorize(self) -> Tuple[np.ndarray, Any]:
+        """
+        See docstring in `ExtensionArray` class in `pandas/core/arrays/base.py`
+        for information about this method.
+        """
+        # TODO return self._tensor, np.nan
+        raise NotImplementedError
```

Comment:
not sure how useful this will be, but I think there should be something here for completness

Reply:
Yeah, you'll probably find that part of the Pandas suite for extension types fails if this method isn't implemented.